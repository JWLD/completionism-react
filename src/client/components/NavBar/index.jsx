import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link, withRouter } from 'react-router-dom'

import { CONTENT } from 'constants/content'
import * as SC from './styled'

class NavBar extends Component {
  renderContentNav() {
    if (!this.props.match.params.category) return null

    const { category, content } = this.props.match.params
    const prevPage = Number(content) - 1
    const nextPage = Number(content) + 1

    return (
      <SC.CentreWrap>
        <SC.NavLink
          to={`/browse/${category}/${prevPage}`}
          className="fa fa-arrow-circle-left"
        />

        <SC.MainTitle>{CONTENT[content].content}</SC.MainTitle>

        <SC.NavLink
          to={`/browse/${category}/${nextPage}`}
          className="fa fa-arrow-circle-right"
        />
      </SC.CentreWrap>
    )
  }

  renderCategorySpan() {
    const { category } = this.props.match.params

    return category && <SC.NavSpan>{category}</SC.NavSpan>
  }

  render() {
    return (
      <SC.NavBar>
        <SC.LeftWrap>
          <SC.NavLink className="fa fa-home" to="/" />

          {this.renderCategorySpan()}
        </SC.LeftWrap>

        {this.renderContentNav()}
      </SC.NavBar>
    )
  }
}

NavBar.propTypes = {
  match: PropTypes.object.isRequired
}

export default withRouter(NavBar)
