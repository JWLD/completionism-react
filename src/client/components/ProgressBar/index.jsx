import React from 'react'
import PropTypes from 'prop-types'

import * as SC from 'ProgressBar/styled'

const ProgressBar = ({ count, height }) => (
  <SC.ProgressBar height={height}>
    <SC.BarFill fill={count} height={height} />
  </SC.ProgressBar>
)

ProgressBar.propTypes = {
  count: PropTypes.number.isRequired,
  height: PropTypes.number
}

ProgressBar.defaultProps = {
  height: 1.5
}

export default ProgressBar
