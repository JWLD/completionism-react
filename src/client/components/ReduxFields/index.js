export { default as InputField } from './InputField';
export { default as SelectBoxField } from './SelectBoxField';
