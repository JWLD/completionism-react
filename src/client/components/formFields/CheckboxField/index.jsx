import React from 'react'
import PropTypes from 'prop-types'

const CheckboxField = ({ input, StyledComponent }) => (
  <StyledComponent {...input} type="checkbox" />
)

CheckboxField.propTypes = {
  input: PropTypes.object.isRequired,
  StyledComponent: PropTypes.func.isRequired
}

export default CheckboxField
