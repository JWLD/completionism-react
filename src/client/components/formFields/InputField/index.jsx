import React from 'react';
import PropTypes from 'prop-types';

const InputField = ({ input, placeholder, StyledComponent }) => (
	<StyledComponent {...input} placeholder={placeholder} />
);

InputField.propTypes = {
	input: PropTypes.object.isRequired,
	placeholder: PropTypes.string.isRequired,
	StyledComponent: PropTypes.func.isRequired
};

export default InputField;
