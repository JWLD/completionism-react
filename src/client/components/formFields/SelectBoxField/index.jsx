import React from 'react';
import PropTypes from 'prop-types';

const SelectBoxField = ({ input, options, placeholder, StyledComponent }) => {
	const placeholderOption = placeholder
		? <option value="">{placeholder}</option>
		: null;

	return (
		<StyledComponent {...input}>
			{placeholderOption}
			{Object.keys(options).map(key => (
				<option key={key} value={key}>
					{options[key]}
				</option>
			))}
		</StyledComponent>
	);
};

SelectBoxField.propTypes = {
	input: PropTypes.object.isRequired,
	options: PropTypes.object.isRequired,
	placeholder: PropTypes.string,
	StyledComponent: PropTypes.func.isRequired
};

SelectBoxField.defaultProps = {
	placeholder: null
};

export default SelectBoxField;
