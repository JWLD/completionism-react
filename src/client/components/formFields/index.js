export { default as CheckboxField } from 'formFields/CheckboxField'
export { default as InputField } from 'formFields/InputField'
export { default as SelectBoxField } from 'formFields/SelectBoxField'
