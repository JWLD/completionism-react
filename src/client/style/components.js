import styled from 'styled-components'

import mixins from 'style/mixins'
import vars from 'style/variables'

export const PageWrap = styled.div`
  ${mixins.flexVertical};

  min-width: 100vw;
  margin-top: ${vars.navBarHeight}rem;
  padding: ${vars.pagePadding}rem;
`

export const BrowseBlock = styled.div`
  ${mixins.flexVertical};

  position: relative;
  width: 100%;
  padding: ${vars.itemListPadding}rem;
  border-radius: 0.4rem;
  background-color: ${vars.mainGrey};

  &:not(:last-child) {
    margin-bottom: 2rem;
  }
`
