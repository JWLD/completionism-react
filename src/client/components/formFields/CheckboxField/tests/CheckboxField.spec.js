import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { CheckboxField } from 'components/formFields'

Enzyme.configure({ adapter: new Adapter() })

const mockProps = {
  input: {},
  StyledComponent: jest.fn()
}

describe('<CheckboxField />', () => {
  it('renders correctly', () => {
    const component = shallow(<CheckboxField {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
