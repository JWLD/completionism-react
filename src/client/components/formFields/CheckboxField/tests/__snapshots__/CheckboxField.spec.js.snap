// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxField /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckboxField
    StyledComponent={[MockFunction]}
    input={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "type": "checkbox",
    },
    "ref": null,
    "rendered": null,
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "type": "checkbox",
      },
      "ref": null,
      "rendered": null,
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
